src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/SnippetTaglet.java

```diff
@@ -0,0 +1,366 @@
+            String msg = writer.configuration().getDocResources()
+                .getText("doclet.snippet.markup", e.getMessage());
+            writer.configuration().getReporter().print(Diagnostic.Kind.ERROR,
+                path, e.getPosition(), e.getPosition(), e.getPosition(), msg);
+            return badSnippet(writer);
+        }
+
+        try {
+            if (externalContent != null) {
+                externalSnippet = parse(externalContent);
+            }
+        } catch (ParseException e) {
+            assert fileObject != null;
+            writer.configuration().getMessages().error(fileObject, e.getPosition(),
+                e.getPosition(), e.getPosition(), "doclet.snippet.markup", e.getMessage());
+            return badSnippet(writer);
+        }
+
+        // the region must be matched at least in one content: it can be matched
+        // in both, but never in none
+        if (regionName != null) {
+            StyledText r1 = null;
+            StyledText r2 = null;
+            if (inlineSnippet != null) {
+                r1 = inlineSnippet.getBookmarkedText(regionName);
+                if (r1 != null) {
+                    inlineSnippet = r1;
+                }
+            }
+            if (externalSnippet != null) {
+                r2 = externalSnippet.getBookmarkedText(regionName);
+                if (r2 != null) {
+                    externalSnippet = r2;
+                }
+            }
+            if (r1 == null && r2 == null) {
+                error(writer, holder, tag, "doclet.snippet.region.not_found", regionName);
+                return badSnippet(writer);
+            }
+        }
+
+        if (inlineSnippet != null) {
+            inlineSnippet = toDisplayForm(inlineSnippet);
+        }
+
+        if (externalSnippet != null) {
+            externalSnippet = toDisplayForm(externalSnippet);
+        }
+
+        if (inlineSnippet != null && externalSnippet != null) {
+            String inlineStr = inlineSnippet.asCharSequence().toString();
+            String externalStr = externalSnippet.asCharSequence().toString();
+            if (!Objects.equals(inlineStr, externalStr)) {
+                error(writer, holder, tag, "doclet.snippet.contents.mismatch", diff(inlineStr, externalStr));
+                // output one above the other
+                return badSnippet(writer);
+            }
+        }
+
+        assert inlineSnippet != null || externalSnippet != null;
+        StyledText text = inlineSnippet != null ? inlineSnippet : externalSnippet;
+
+        return writer.snippetTagOutput(holder, snippetTag, text);
+    }
+
+    /*
+     * Maybe there's a case for implementing a proper (or at least more helpful)
+     * diff view, but for now simply outputting both sides of a hybrid snippet
+     * would do. A user could then use a diff tool of their choice to compare
+     * those sides.
```

Comment:
For discussion ... a partial solution that may be good enough is to identify the position (e.g. line+offset) of the first character that is different.